spec.md

```diff
@@ -572,217 +572,152 @@ See [@sec:mem] for more details.
+They behave like zero when they are extended to a positive width integer type.
 
-``` firrtl
-wire one_u : UInt<1>
-connect one_u, UInt<1>(0)
-wire one_s : SInt<1>
-connect one_s, SInt<1>(0)
-```
+Integer types also have the inferred forms: `UInt`{.firrtl} and `SInt`{.firrtl} (see [@sec:width-inference]).
 
 ### Clock Type
 
-The clock type is used to describe wires and ports meant for carrying clock signals.
-The usage of components with clock types are restricted.
-Clock signals cannot be used in most primitive operations, and clock signals can only be connected to components that have been declared with the clock type.
-
-The clock type is specified as follows:
-
-``` firrtl
-Clock
-```
-
-### Reset Type
-
-The uninferred `Reset`{.firrtl} type is either inferred to `UInt<1>`{.firrtl} (synchronous reset) or `AsyncReset`{.firrtl} (asynchronous reset) during compilation.
+Clocks require special physical considerations in hardware.
+FIRRTL defines the `Clock`{.firrtl} type to track clocks throughout a design.
+All registers are linked to a clock (see [@sec:registers]).
 
-``` firrtl
-Reset ; inferred type
-AsyncReset
-```
-
-Synchronous resets used in registers will be mapped to a hardware description language representation for synchronous resets.
-
-The following example shows an uninferred reset that will get inferred to a synchronous reset.
+### Reset Types
 
-``` firrtl
-input a : UInt<1>
-wire reset : Reset
-connect reset, a
-```
+Once a circuit is powered on, it may require an explicit reset in order to put it into a known state.
+For this, we use a reset type.
+In FIRRTL, we have the option of using both synchronous or asynchronous resets.
 
-After reset inference, `reset`{.firrtl} is inferred to the synchronous `UInt<1>`{.firrtl} type:
+The synchronous reset type is simply a 1-bit unsigned integer: `UInt<1>`{.firrtl}.
+The asynchronous reset type is `AsyncReset`{.firrtl}.
 
-``` firrtl
-input a : UInt<1>
-wire reset : UInt<1>
-connect reset, a
-```
+Registers may be declared linked to a reset (see [@sec:registers-with-reset]).
 
-Asynchronous resets used in registers will be mapped to a hardware description language representation for asynchronous resets.
+The reset types also have the inferred form: `Reset`{.firrtl} (see [@sec:reset-inference]).
 
-The following example demonstrates usage of an asynchronous reset.
+### Analog Type
 
-``` firrtl
-input clock : Clock
-input reset : AsyncReset
-input x : UInt<8>
-regreset y : UInt<8>, clock, reset, UInt(123)
-; ...
-```
+The analog type specifies that a wire or port can be attached to multiple drivers.
+`Analog`{.firrtl} cannot be used as part of the type of a node or register, nor can it be used as part of the datatype of a memory.
+In this respect, it is similar to how `inout`{.firrtl} ports are used in Verilog, and FIRRTL analog signals are often used to interface with external Verilog or VHDL IP.
```

Comment:
This is fine as commentary here. However, there are a few things missing/unclear: (1) How is this represented in the ABI? (There is a missing ABI section on Analog lowering.) (2) Given the ABI, Analog does lower to `inout`, therefore, is it appropriate to say `Analog` is "similar to how inout ports are used in Verilog"?

Reply:
This is the original copy.